app/controllers/sessions_controller.rb

```diff
@@ -27,4 +25,16 @@ def check_remember
       forget @user
     end
   end
+
+  def activated
+    if @user.activated
+      log_in @user
+      check_remember
+      redirect_back_or @user
+    else
+      message = t "login_activate_message"
+      flash[:warning] = message
```

Comment:
ko cần khai báo biến dư thừa nhé
```suggestion
      flash[:warning] = t "login_activate_message"
```